Increase IO surface pool size on macOS

rniwa opened the pull request:
#### 71061fa2508a5676a4b0506a8dfcc5a69277e0d7
<pre>
Increase IO surface pool size on macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=264912">https://bugs.webkit.org/show_bug.cgi?id=264912</a>

Reviewed by Simon Fraser.

Increase the IO surface pool size from 64MB to 256MB on macOS.
This increases the cache hit rate from ~30% to ~80% and improves Speedometer 3 score by ~0.7%.

* Source/WebCore/platform/graphics/cg/IOSurfacePool.h:

Canonical link: <a href="https://commits.webkit.org/270817@main">https://commits.webkit.org/270817@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7f4ce955696114a46488c322e361e50575c06ed6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26496 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5111 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27743 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28686 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24232 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26903 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6912 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2519 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/24190 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26756 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3938 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22750 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3465 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/3513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23729 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29190 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24152 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24130 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29781 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3547 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/1746 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27675 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4981 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4023 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3427 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3872 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/platform/graphics/cg/IOSurfacePool.h:
```diff
@@ -78,7 +78,11 @@ class IOSurfacePool : public ThreadSafeRefCounted<IOSurfacePool> {
     typedef HashMap<IOSurface*, CachedSurfaceDetails> CachedSurfaceDetailsMap;
 
+#if PLATFORM(MAC)
+    static constexpr size_t defaultMaximumBytesCached { 1024 * 1024 * 256 };
```
We must have a BYTES_PER_MEGABYTE somewhere.

rniwa replied:
Hm... I don't find it anywhere.

Constellation replied:
We have `MB` in StdLibExtras.h, so you can use `256 * MB`.

rniwa replied:
oh neat.

rniwa committed:
Increase IO surface pool size on macOS
https://bugs.webkit.org/show_bug.cgi?id=264912

Reviewed by Simon Fraser.

Increase the IO surface pool size from 64MB to 256MB on macOS.
This increases the cache hit rate from ~30% to ~80% and improves Speedometer 3 score by ~0.7%.

* Source/WebCore/platform/graphics/cg/IOSurfacePool.h:

Canonical link: https://commits.webkit.org/270817@main
